akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -719,7 +721,11 @@ abstract class ShardCoordinator(
             case Some(shards) =>
               log.debug("Graceful shutdown of region [{}] with shards [{}]", region, shards)
               gracefulShutdownInProgress += region
-              continueRebalance(shards.toSet)
+
+              // only for rebalance if we have at least one active region
+              if (calculateActiveRegions() > 1)
+                continueRebalance(shards.toSet)
+
```

Comment:
I initially thought about simply check if there is more than one region, but then thought about the following scenario:

* Cluster with two regions
* both regions start to shutdown
* `GracefulShutdownReq` from first region is received, coordinator runs a rebalance (we can't avoid this one)
* second region sends a `GracefulShutdownReq`, first region is not yet gone, but it makes no sense to run a rebalance. So we skip it.